### DISCUSSION TITLE
Keybind that allows for literal editing of interactive rebase files

### DISCUSSION THREAD

**Discussion by LucianoLaratelli** [magit/magit#4380] [Support] [answered]
Hello,

I love `magit`'s interactive rebase feature; it is delightful and I am happy every time I get to use it. 

One thing I would like to be able to do is reorder multiple commits at once. 
Right now, I can reorder a commit with `g{j,k}`. This works fine for single commits, but if I have a block of commits I can't select them and then `g{j,k}` them. Ideally, this would be supported inside `magit`, but I recognize that may not be possible.

A friend came up with this workaround, which seems to work:

1. switch to `fundamental-mode`
2. toggle `read-only-mode`
3. At this point, you can move things around easily, because it's just text
4. Switch back to `git-rebase-mode`

It would be nice if `magit` had a keybinding that dropped one into directly editing the rebase file and then (when in that mode, maybe `magit-rebase-literal-mode` or something) another keybinding that got one back to the normal rebase mode.

Thank you all for your hard work on this wonderful project!

**Comment by tarsius** [comment, marked as answer]
> but if I have a block of commits I can't select them and then `g{j,k}` them.

Yes you can; just mark them using the region before using the movement commands.

**Reply by LucianoLaratelli** [reply]
Oh interesting. This may be a bug in EVIL or Doom then, I'll look further there. Thanks Jonas :)
